From 3ece54b0e34e45804e6d344d2d953b6f716cd6a9 Mon Sep 17 00:00:00 2001
From: "haidong.zheng" <haidong.zheng@nxp.com>
Date: Mon, 22 Jul 2024 17:48:11 +0800
Subject: [PATCH 20/56] LF-15588: arm64: dts: freescale: Add ld mode specific
 dts for frdm-imx93

Add ld mode specific dts to limit the flexspi & usdhc clock to
ld mode by default.

Signed-off-by: haidong.zheng <haidong.zheng@nxp.com>
Reviewd-by: Peng Fan <peng.fan@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../dts/freescale/imx93-11x11-frdm-ld.dts     | 34 +++++++++++++++++++
 2 files changed, 35 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ld.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index fe384b93b97f..0b45c9cc3f35 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -495,7 +495,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-evk.dtb \
 			  imx93-11x11-evk-8mic-reve.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-frdm.dtb \
-			  imx93-11x11-frdm-aud-hat.dtb
+			  imx93-11x11-frdm-aud-hat.dtb imx93-11x11-frdm-ld.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx91-11x11-evk.dtb \
 			  imx91-11x11-evk-flexspi-m2.dtb imx91-11x11-evk-flexspi-nand-m2.dtb \
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ld.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ld.dts
new file mode 100644
index 000000000000..adf9f4809d32
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-ld.dts
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx93-11x11-frdm.dts"
+
+&flexspi1 {
+	assigned-clocks = <&clk IMX93_CLK_FLEXSPI1>;
+	assigned-clock-parents = <&clk IMX93_CLK_SYS_PLL_PFD1>;
+	assigned-clock-rates = <266000000>;
+};
+
+&lpm {
+	ld-mode-enabled;
+};
+
+&usdhc1 {
+	assigned-clocks = <&clk IMX93_CLK_USDHC1>;
+	assigned-clock-parents = <&clk IMX93_CLK_SYS_PLL_PFD1>;
+	assigned-clock-rates = <266000000>;
+};
+
+&usdhc2 {
+	assigned-clocks = <&clk IMX93_CLK_USDHC2>;
+	assigned-clock-parents = <&clk IMX93_CLK_SYS_PLL_PFD1>;
+	assigned-clock-rates = <266000000>;
+};
+
+&usdhc3 {
+	assigned-clocks = <&clk IMX93_CLK_USDHC3>;
+	assigned-clock-parents = <&clk IMX93_CLK_SYS_PLL_PFD1>;
+	assigned-clock-rates = <266000000>;
+};
-- 
2.34.1

